nltk/stem/porter.py

```diff
@@ -652,13 +652,16 @@ def stem(self, word):
         stem = word.lower()
 
         if self.mode == self.NLTK_EXTENSIONS and word in self.pool:
-            return self.pool[word]
+            return self.pool[stem]
 
         if self.mode != self.ORIGINAL_ALGORITHM and len(word) <= 2:
             # With this line, strings of length 1 or 2 don't go through
             # the stemming process, although no mention is made of this
             # in the published algorithm.
-            return word
+            if word.isupper()==True:
+                return word
+            else:
+                return stem
```

Comment:
Perhaps the ternary operator here is appropriate:

```
return word if word.isupper() else stem
```

Also, when checking for truthiness of string, `if word.isupper()` should be sufficient without the explicity `==True` =)

Reply:
Sure. I'll fix it.